src/components/editor/actions/CloseButton.jsx

```diff
@@ -13,8 +13,9 @@ const CloseButton = (props) => {
   const dispatch = useDispatch()
   const history = useHistory()
 
-  const resourceKey = useSelector((state) => currentResourceKey(state))
-  const resourceHasChanged = useSelector((state) => resourceHasChangesSinceLastSave(state))
+  let resourceKey = useSelector((state) => currentResourceKey(state))
+  if (props.resourceKey) { resourceKey = props.resourceKey }
```

Comment:
How would you feel about reworking this as follows:

```javascript
const resourceKey = props.resourceKey ? props.resourceKey : useSelector((state) => currentResourceKey(state))
```

Reply:
Originally I was doing something similar but got an `eslint` error that `useSelector` needed to be a deterministic function call.

Reply:
OK, sounds good!